test/framework/deployment_helpers.go

```diff
@@ -179,51 +183,112 @@ type watchPodLogsInput struct {
 // retrieving the deployment, its pods, or setting up a log file. If there is an error with the log streaming itself,
 // that does not cause the test to fail.
 func watchPodLogs(ctx context.Context, input watchPodLogsInput) {
-	pods := &corev1.PodList{}
-	Expect(input.GetLister.List(ctx, pods, client.InNamespace(input.Namespace), client.MatchingLabels(input.Labels))).To(Succeed(), "Failed to list Pods for deployment %s", klog.KRef(input.Namespace, input.DeploymentName))
+	// Create informer to watch for pods matching input.
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(
+		input.ClientSet,
+		10*time.Minute,
+		informers.WithNamespace(input.Namespace),
+		informers.WithTweakListOptions(func(opt *metav1.ListOptions) {
+			opt.LabelSelector = metav1.FormatLabelSelector(&metav1.LabelSelector{MatchLabels: input.Labels})
+		}),
+	)
+
+	podInformer := informerFactory.Core().V1().Pods().Informer()
+	_, err := podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
q: should we also stop streaming if the pods goes away?

Reply:
You mean: stop streaming if a deployment goes away?!

Stop streaming for a pod is already handled in `streamPodLogs`, which will not continue trying to stream if `...Pods(..).Get(..)` is a `IsNotFound` error during retries.